File: ruby/number2words/lib/number2words/number.rb
module Number2Words
  class Number
    def initialize value
      @value = value
    end

    attr_reader :value

    def to_words
      Digits.of(self).to_words
    end

    def to_a
      @value.to_s
        .chars
        .map(&:to_i)
        .reverse
    end
  end
end

File: CSharp/Number2WordsCSharp/Number2Words.Core/Number.cs
using System.Linq;

namespace Number2Words.Core
{
    public class Number
    {
        private int Value { get; set; }

        public Number(int value)
        {
            Value = value;
        }

        public string ToWords()
        {
            return ToDigits().ToWords();
        }

        private Digits ToDigits()
        {
            return Value
                .ToString()
                .ToCharArray()
                .Reverse()
                .Select(c => int.Parse(c.ToString()))
                .ToDigits();
        }
    }
}